src/plugins/controls/public/control_group/editor/data_control_editor_tools.ts

```diff
@@ -30,20 +30,30 @@ const loadFieldRegistryFromDataView = async (
   const controlFactories = getControlTypes().map(
     (controlType) => getControlFactory(controlType) as IEditableControlFactory
   );
-  const fieldRegistry: DataControlFieldRegistry = dataView.fields
-    .getAll()
-    .reduce((registry, field) => {
-      const test: DataControlField = { field, compatibleControlTypes: [] };
-      for (const factory of controlFactories) {
-        if (factory.isFieldCompatible) {
-          factory.isFieldCompatible(test);
-        }
+  const fieldRegistry: DataControlFieldRegistry = {};
+  await Promise.all(
```

Comment:
One other performance consideration is to use `for` loop for iterating through large collections. Each time `map` is run, a new function is created and this creates a large garbage collection penalty.

Reply:
True! I bet we could increase this performance even further, cause the code snipped that I suggested also uses `map` unnecessarily.

Reply:
Changed to `forEach` in 9d29b33f266f06cdf92daf7d053a7805c472c347